xwiki-platform-core/xwiki-platform-activitystream/xwiki-platform-activitystream-api/src/main/java/com/xpn/xwiki/plugin/activitystream/eventstreambridge/BridgeEventStream.java

```diff
@@ -80,11 +94,23 @@ public void deleteEvent(Event e)
             XWikiContext context = getXWikiContext();
             ActivityStreamPlugin plugin = getPlugin(context);
             plugin.getActivityStream().deleteActivityEvent(eventConverter.convertEventToActivity(e), context);
+            this.sendEventStreamEvent(new EventStreamDeletedEvent(e));
         } catch (ActivityStreamException ex) {
             // Unlikely; nothing we can do
         }
     }
 
+    private void sendEventStreamEvent(AbstractEventStreamEvent eventStreamEvent)
+    {
+        // In order to avoid infinite loop caused by observation events triggering event stream events that are
+        // themselves triggering new observation events …, we add a new property to the execution context. Therefore,
+        // an observation event declared out of an event stream event can only be triggered once.
+        if (!this.execution.getContext().hasProperty(EVENT_CONTEXT_LOCK_PROPERTY)) {
+            this.execution.getContext().newProperty(EVENT_CONTEXT_LOCK_PROPERTY).declare();
```

Comment:
@tmortagne: Can you confirm that the execution context has its properties flushed at the end of a request ?
If not, I’ll have to add something else here.